crypto/bio/bio.c

```diff
@@ -299,13 +308,42 @@ int BIO_write_all(BIO *bio, const void *data, size_t len) {
 }
 
 int BIO_puts(BIO *bio, const char *in) {
-  size_t len = strlen(in);
+  const size_t len = strlen(in);
```

Comment:
You should be able to move this into else code block on lines 337-339 if you implement the other feedback below.